test/server/proxy-test.js

```diff
@@ -604,5 +611,25 @@ describe('Proxy', function () {
                 done();
             });
         });
+
+        it('Should recalculate content length after the upload info is injected (GH-195)', function (done) {
+            var src      = newLineReplacer(fs.readFileSync('test/server/data/upload/src.formdata'));
+            var expected = newLineReplacer(fs.readFileSync('test/server/data/upload/expected.formdata'));
+
+            var options = {
+                url:     proxy.openSession('http://127.0.0.1:2000/GH-195/content-length', session),
+                method:  'POST',
+                body:    src,
+                headers: {
+                    'content-type': 'multipart/form-data; boundary=separator',
+                    'accept':       'text/plain;q=0.9,*!/!*;q=0.8'
+                }
+            };
+
+            request(options, function (err, res, body) {
+                expect(body).eql(expected.length.toString());
```

Comment:
Once again: why not send back the body and compare it with `expected`? Why we compare just length and not content?